contracts/src/v0.8/keystone/KeystoneFeedsConsumer.sol

```diff
@@ -2,27 +2,92 @@
 pragma solidity ^0.8.19;
 
 import {IReceiver} from "./interfaces/IReceiver.sol";
+import {ConfirmedOwner} from "../shared/access/ConfirmedOwner.sol";
 
-contract KeystoneFeedsConsumer is IReceiver {
-  event MessageReceived(bytes32 indexed workflowId, address indexed workflowOwner, uint256 nReports);
+contract KeystoneFeedsConsumer is IReceiver, ConfirmedOwner {
   event FeedReceived(bytes32 indexed feedId, uint256 price, uint64 timestamp);
 
-  constructor() {}
+  error UnauthorizedSender(address sender);
+  error UnauthorizedWorkflowOwner(address workflowOwner);
+  error UnauthorizedWorkflowName(bytes10 workflowName);
+
+  constructor() ConfirmedOwner(msg.sender) {}
 
   struct FeedReport {
-    bytes32 FeedID;
+    bytes32 FeedId;
     uint256 Price;
```

Comment:
this could be changed to a `uint224` and `uint32` for Price and Timestamp respectively to save the additional storage slot, and thus ~22k gas with every report. A uint32 is more than enough for a timestamp (the year 2106)